sessionctx/variable/tidb_vars.go

```diff
@@ -228,6 +234,7 @@ const (
 	DefTiDBGeneralLog                = 0
 	DefTiDBRetryLimit                = 10
 	DefTiDBDisableTxnAutoRetry       = false
+	DefTiDBDeferConstraintCheck      = true
```

Comment:
I prefer to use `false` as default value of a variable.
Did you forget to set this value in sessionVars ? @jackysp

Reply:
If false is the preferred default, the name would need to change.  Perhaps something like `synchronous_constraint_check`?  I do prefer the word `defer` or `delay` though.  `delay` is used in MySQL for non-safe use cases though (myisam's delay-key-write), which is why I suggested defer.

Reply:
The name 'Deferrable constraints' is also common in other DBs.  For example: https://www.postgresql.org/docs/9.1/static/sql-set-constraints.html

Reply:
Yeah, we can change a variable name. @morgo @jackysp 
It's golang idiom to use `false` as default value, because when you forget to initialize a variable, it will be set to false automatically in golang, so use `false` is less likely to get trapped.

Reply:
`tidb_disable_deferred_constraints` ?

Reply:
@morgo I think a double negative is good for the user to use.
@tiancaiamao I think it's not about the golang's idiom. If it is a bug when the default value is true, you could point it.

Reply:
`tidb_disable_deferred_constraints` sounds good

Reply:
@lilin90 we need some help of the name issue.

Reply:
I don't think we can let golang zero values dictate how variable names end up facing the user. Also, consider that we can decide to change the default for a variable in the future. You can use a pointer or an enumeration in go if that helps, but we probably need a better abstraction for our variables.

Reply:
I agree with @lilin90 's suggestion.